File: biopython_rewrite/report_summary.py
#!/usr/bin/env python3
# Create tiny HTML summary from TSV stats
import sys
if len(sys.argv)<2:
    print('Usage: report_summary.py stats.tsv'); sys.exit(1)
print('<html><body><table>')
with open(sys.argv[1]) as fh:
    for line in fh:
        cols = line.strip().split('\t')
        print('<tr>' + ''.join(f'<td>{c}</td>' for c in cols) + '</tr>')
print('</table></body></html>')

File: legacy_perl/report_summary.pl
#!/usr/bin/env perl
# Create a tiny HTML summary report from tabular stats
use strict; use warnings;
my $in = shift or die "Usage: $0 stats.tsv\n";
open my $fh,'<',$in or die $!;
print "<html><body><table>\n";
while(<$fh>){
    chomp; my @c=split/\t/;
    print "<tr>" . join('', map {"<td>$_</td>"} @c) . "</tr>\n";
}
print "</table></body></html>\n";
